### PR TITLE
5188 integrate r15

### PR DESCRIPTION
**Pull request by weskubo-cgi**
Integrated with Maximus.
Update properties.
Update unit tests.
UI updates.
Update testcafe tests.

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java [hunk 1]
```diff
@@ -142,8 +136,6 @@ public WebClient hibcWebClient() throws HNWebException {
                 .filter(logRequest())
                 .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN_VALUE) 
                 .defaultHeader(HttpHeaders.ACCEPT, MediaType.ALL_VALUE)
-                // TODO (weskubo-cgi) Remove default headers and corresponding properties
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java:145]: Doesn't the R15 still require basic Auth? 
Also don't the R15 and R50 share an endpoint? Should these just be the same WebClient?

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java [hunk 2]
```diff
@@ -65,11 +65,18 @@
 
 			logger.info("checkEligibility response: {} ", checkEligibilityResponse);
 			return response;	
+		} catch (HNWebException hwe) {
+			switch (hwe.getType()) {
+			case DOWNSTREAM_FAILURE:
+				throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, hwe.getMessage(), hwe);
+			default:
+				throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Bad /lookup-phn request", hwe);
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java:73]: Copy paste errors here should be /check-eligibility

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/service/EligibilityService.java [hunk 3]
```diff
@@ -78,16 +76,37 @@
...
+			throw new HNWebException(ExceptionType.DOWNSTREAM_FAILURE);
+		}
+
+		return parseResponse(response.getBody(), "R15");
+	}
+	
+	/**
+	 * Checks for MSP Coverage Status based on the E45 request.
+	 * @param e45
+	 * @return
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/service/EligibilityService.java:95]: Javadoc is inconsistent in this file
Some methods have Javadoc, some are missing @return tag descriptions like here
Some are missing @param and @throws tag descriptions.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/converter/hl7v2/BaseV2Converter.java [hunk 4]
```diff
@@ -67,7 +77,30 @@ protected void populateIN1(IN1 in1, LocalDate planEffectiveDate) throws HL7Excep
 	protected void populateIN1(IN1 in1, LocalDate planEffectiveDate, LocalDate planCancellationDate, String groupNumber, String groupMemebrNumber, String immigrationCode ) throws HL7Exception {
 		V2MessageUtil.setIn1Values(in1, groupNumber, groupMemebrNumber, immigrationCode, dateOnlyFormatter.format(planEffectiveDate), dateOnlyFormatter.format(planCancellationDate));
 	}
-	
-	
-	
+
+	protected void mapErrorValues(Terser terser, Message message, BaseResponse response) throws HL7Exception {
```
Comment (tschia) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/converter/hl7v2/BaseV2Converter.java:81]: Message message parameter isn't actually used

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java (hunk 1 above)
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java:145]: We discovered during testing that E45/R15 and R50 are actually set up as different user accounts for basic auth. So to support this the auth was decoupled from the webclient. I established the pattern in E45/R15 and Dave will update the code so there isn't an R50 specific webclient or service methods...but just it's own set of credentials.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java (hunk 2 above)
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java:73]: Fixed. Plan is to move to a global ExceptionHandler eventually as the exception handling code is getting duplicated and really all our controllers follow a similar pattern.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/service/EligibilityService.java (hunk 3 above)
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/service/EligibilityService.java:95]: Updated javadoc for consistency.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/converter/hl7v2/BaseV2Converter.java (hunk 4 above)
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/converter/hl7v2/BaseV2Converter.java:81]: Removed.

**Review by tschia — APPROVED** [review]

**Pull request merged**